pkg/agents/team7/agent1/agent.go

```diff
@@ -79,99 +96,495 @@ func New(baseAgent *infra.Base) (infra.Agent, error) {
 			agreeableness:     agreeableness,
 			neuroticism:       neuroticism,
 		},
-		greediness:   100 - agreeableness,
-		kindness:     agreeableness,
-		daysHungry:   0,
-		seenPlatform: false,
-		foodEaten:    0,
-		prevHP:       100,
-		prevFloors:   []int{},
+		behaviour: CurrentBehaviour{
+			greediness:     ((100 - agreeableness) / 2) + 25,
+			kindness:       agreeableness,
+			responsiveness: (agreeableness + openness + extraversion) / 3,
+		},
+		opMem: OperationalMemory{
+			orderPrevFloors:   []int{},
+			prevFloors:        prevFloors,
+			currentDayonFloor: 0,
+			numOfDays:         0,
+			avgFood:           0,
+			currentFloorRisk:  0,
+			daysHungry:        0,
+			seenPlatform:      false,
+			prevAge:           -1,
+			prevHP:            100,
+			foodEaten:         0,
+			messagesSent:      false,
+			msg1Sent:          false,
+			msg2Sent:          false,
+			msg3Sent:          false,
+			recievedReq:       false,
+			takeFood:          0,
+			leaveFood:         0,
+		},
 	}, nil
 }
 
 func (a *CustomAgent7) Run() {
 
-	a.Log("Team7Agent1 reporting status:", infra.Fields{"floor": a.Floor(), "hp": a.HP(), "greed": a.greediness, "kind": a.kindness, "aggr": a.personality.agreeableness})
+	// a.Log("Team7Agent1 reporting status:", infra.Fields{"floor": a.Floor(), "hp": a.HP(), "greed": a.behaviour.greediness, "kind": a.behaviour.kindness, "aggr": a.personality.agreeableness})
+
+	// Operational Variables
+	UserID := a.ID()
+	currentHP := a.HP()
+	daysCritical := a.DaysAtCritical()
+	healthInfo := a.HealthInfo()
+	currentFloor := a.Floor()
+
+	// Only occurs on a floor change
+	if len(a.opMem.orderPrevFloors) == 0 || a.opMem.orderPrevFloors[len(a.opMem.orderPrevFloors)-1] != currentFloor { //If day 1 or floor change
+
+		a.Log("checking floor change........................................................................................")
+		if a.Age() == 0 {
+			a.Log("Personality Makeup:", infra.Fields{"A": a.personality.agreeableness, "C": a.personality.conscientiousness, "E": a.personality.extraversion, "N": a.personality.neuroticism, "O": a.personality.openness})
+		}
+
+		//Reset Variables on Floor Change
+		a.opMem.currentDayonFloor = 1 // reset currentDay counter
+		a.opMem.recievedReq = false   // reset Requests
+		a.opMem.takeFood = 0
```

Comment:
Something to take into account if this is later used in decisions, how do you differentiate between going to eat 0 and not having decided what to eat?

Reply:
Same with leave Food